src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -645,8 +645,23 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     // If the theme asks for Mica, then drop out our background, so that we
     // can have mica too.
     void MainPage::_UpdateBackgroundForMica()
-
     {
+        // DWMWA_SYSTEMBACKDROP_TYPE, which we use for mica, is only available
+        // on builds >=2261
+        static const bool isMicaAvailable = []() -> bool {
+            OSVERSIONINFOEXW osver{};
+            osver.dwOSVersionInfoSize = sizeof(osver);
+            osver.dwBuildNumber = 22621;
+            DWORDLONG dwlConditionMask = 0;
+            VER_SET_CONDITION(dwlConditionMask, VER_BUILDNUMBER, VER_GREATER_EQUAL);
+            return VerifyVersionInfoW(&osver, VER_BUILDNUMBER, dwlConditionMask) != FALSE;
+        }();
+
+        if (!isMicaAvailable)
+        {
+            return;
+        }

```

Comment:
I believe it'd be better long-term to come up with a way how we can get to the return value of `DwmSetWindowAttribute` in `IslandWindow::UseMica` instead.

Reply:
I don't disagree, but that starts to be a little tricky from a layering perspective. The SUI dll doesn't link against anything at the window level, so it can't check that itself.

So instead we'd have to plumb the result back down from the window->app->SUI, to say "yes, you requested mica _and you got it_". That's doable, I suppose

Reply:
Consider using, ah, well...

We pass the HWND down into the settings UI so that we can parent the file dialogs appropriately.

You could use `DwmGetWindowAttribute` on that HWND to see whether Mica is, in fact, enabled...